librz/core/cmd/cmd_open.c

```diff
@@ -1251,3 +778,322 @@ RZ_IPI RzCmdStatus rz_open_binary_reload_handler(RzCore *core, int argc, const c
+
+err:
+	free(data);
+	return res;
+}
+
+static RzCmdStatus open_nobin_file(RzCore *core, const char *uri, ut64 addr, int perms) {
+	if (!strcmp(uri, "=")) {
+		uri = "malloc://512";
+	}
+
+	RzIODesc *desc = rz_io_open_at(core->io, uri, perms, 0644, addr, NULL);
+	if (!desc || desc->fd == -1) {
+		RZ_LOG_ERROR("Cannot open '%s' at %" PFMT64x ".\n", uri, addr);
+		return RZ_CMD_STATUS_ERROR;
+	}
+
+	core->num->value = desc->fd;
+	rz_core_block_read(core);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_open_nobin_handler(RzCore *core, int argc, const char **argv) {
+	ut64 addr = argc > 2 ? rz_num_math(core->num, argv[2]) : 0;
+	int perms = argc > 3 ? rz_str_rwx(argv[3]) : RZ_PERM_R;
+	return open_nobin_file(core, argv[1], addr, perms);
+}
+
+RZ_IPI RzCmdStatus rz_open_nobin_write_handler(RzCore *core, int argc, const char **argv) {
+	ut64 addr = argc > 2 ? rz_num_math(core->num, argv[2]) : 0;
+	int perms = argc > 3 ? rz_str_rwx(argv[3]) : RZ_PERM_RW;
+	return open_nobin_file(core, argv[1], addr, perms);
+}
+
+RZ_IPI RzCmdStatus rz_reopen_handler(RzCore *core, int argc, const char **argv) {
+	int fd;
+	if (argc > 1) {
+		fd = (int)rz_num_math(NULL, argv[1]);
+	} else {
+		if (!core->io || !core->io->desc) {
+			RZ_LOG_ERROR("Cannot find current file.\n");
+			return RZ_CMD_STATUS_ERROR;
+		}
+		fd = core->io->desc->fd;
+	}
+	rz_core_io_file_open(core, fd);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_reopen_write_handler(RzCore *core, int argc, const char **argv) {
+	int fd;
+	int perms = RZ_PERM_RW;
+	if (argc > 1) {
+		fd = (int)rz_num_math(NULL, argv[1]);
+	} else {
+		if (!core->io || !core->io->desc) {
+			RZ_LOG_ERROR("Cannot find current file.\n");
+			return RZ_CMD_STATUS_ERROR;
+		}
+		fd = core->io->desc->fd;
+		perms |= core->io->desc->perm;
+	}
+	rz_core_io_file_reopen(core, fd, perms);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_reopen_binary_handler(RzCore *core, int argc, const char **argv) {
+	rz_core_file_reopen(core, argv[1], 0, 2);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_reopen_core_handler(RzCore *core, int argc, const char **argv) {
+	if (!core->io || !core->io->desc) {
+		RZ_LOG_ERROR("Could not find current file\n");
+		return RZ_CMD_STATUS_ERROR;
+	}
+
+	return open_core_file(core, core->io->desc->uri);
+}
+
+RZ_IPI RzCmdStatus rz_reopen_debug_handler(RzCore *core, int argc, const char **argv) {
+	// TODO: this is bad as we force ourselves to convert arguments to strings.
+	//       There should be an API to reopen a file in debug mode and directly
+	//       pass args to it.
+	char **args = RZ_NEWS(char *, argc - 1);
+	int i;
+	for (i = 1; i < argc; i++) {
+		char *t = rz_cmd_escape_arg(argv[i], RZ_CMD_ESCAPE_DOUBLE_QUOTED_ARG);
+		args[i - 1] = rz_str_newf("\"%s\"", t);
+		free(t);
+	}
+	char *args_str = rz_str_array_join((const char **)args, argc - 1, " ");
+	for (i = 0; i < argc - 1; i++) {
+		free(args[i]);
+	}
+	free(args);
+	rz_core_file_reopen_debug(core, args_str);
```

Comment:
Check for return and handle the errors

Reply:
`RZ_API void rz_core_file_reopen_debug(RzCore *core, const char *args)`